src/perception/camera_object_detection/camera_object_detection/yolov8_detection.py

```diff
@@ -90,7 +101,51 @@ def __init__(self):
         self.get_logger().info(
             f"Successfully created node listening on camera topic: {self.camera_topic}...")
 
-    def preprocess_image(self, cv_image):
+    def crop_image(self, cv_image):
+        if self.crop_mode == "LetterBox":
+            img = LetterBox(self.image_size, stride=self.stride)(
+                image=cv_image)
+        elif self.crop_mode == "CenterCrop":
+            img = CenterCrop(self.image_size)(cv_image)
+        else:
+            raise Exception(
+                "Invalid crop mode, please choose either 'LetterBox' or 'CenterCrop'!")
+
+        return img
+
+    def convert_bboxes_to_orig_frame(self, bbox):
+        """
+        Converts bounding box coordinates from the scaled image frame back to the original image frame.
+
+        This function takes into account the original image dimensions and the scaling method used 
+        (either "LetterBox" or "CenterCrop") to accurately map the bounding box coordinates back to 
+        their original positions in the original image.
+
+        Parameters:
+        bbox (list): A list containing the bounding box coordinates in the format [x1, y1, w1, h1] 
+                    in the scaled image frame.
+
+        Returns:
+        list: A list containing the bounding box coordinates in the format [x1, y1, w1, h1] 
+            in the original image frame.
+        
+        """
+        width_scale = self.orig_image_width / self.image_size
+        height_scale = self.orig_image_height / self.image_size
+        if self.crop_mode == "LetterBox":
+            translation = (self.image_size - self.orig_image_height / width_scale) / 2
+            return [bbox[0] * width_scale,
+                    (bbox[1] - translation) * width_scale,
+                    bbox[2] * width_scale,
+                    bbox[3] * width_scale]
+        elif self.crop_mode == "CenterCrop":
+            translation = (self.orig_image_width / height_scale - self.image_size) / 2
+            return [(bbox[0] + translation) * height_scale,
+                    bbox[1] * height_scale,
+                    bbox[2] * height_scale,
+                    bbox[3] * height_scale]
+
+    def crop_and_convert_to_tensor(self, cv_image):
```

Comment:
maybe we can move these functions into a shared `perception_utils` folder, if these will be used by other nodes like traffic light detection

Lane detection and semantic segmentation will probably use these but are in C++ so we will need to reimplement them in C++ as well

Reply:
this might make docker building a bit tricky, plz keep utils within a single container (ie. object detection utils, not a perception-wide utils)

Reply:
price to pay for modularity

Reply:
IMO Don't need to prematurely modularize, code is ~300 lines right now, once the codebase grows we can split it into different files. Hurts readability to have the logic spread out everywhere